tools/cpp/unix_cc_configure.bzl

```diff
@@ -54,9 +54,41 @@ def _build_tool_path(d):
   return "\n".join(lines)
 
 
+def _get_non_empty_string_from_env(repository_ctx, env_var):
+  tool_path = repository_ctx.os.environ.get(env_var)
+  if not tool_path:
+    return None
+
+  # This is disallowing paths, that have a whitespace on either end
+  tool_path = tool_path.strip()
+
+  if not tool_path:
+    return None # Only return non-empty paths
+  return str(tool_path)
+
+
+def _get_tool_path(repository_ctx, file_name, env_var, default):
+  """Compute the path to a tool"""
+  # Do not use default attribute here since which is an expensive operation
+  tool_override = _get_non_empty_string_from_env(repository_ctx, env_var)
+  if tool_override:
+    return tool_override
+  return which(repository_ctx, file_name, default)
+
+
 def _get_tool_paths(repository_ctx, darwin, cc):
   """Compute the path to the various tools. Doesn't %-escape the result!"""
-  return {k: which(repository_ctx, k, "/usr/bin/" + k)
+
+  if darwin:
```

Comment:
Why use ar on darwin by default? CppActionConfigs.java assume that libtool is used on darwin, so you need to add archiver_flags feature to crosstool.tpl if you insist on ar. But I thought that libtool is favoured over ar on darwin.

Reply:
> Why use ar on darwin by default?
Because the code before was doing so.
Besides isn't _libtool_ more of a gnuish tool, which is kind of "alien" to darwin?

Reply:
I'm sorry I don't follow - Line 71: Use libtool if darwin else ar. This is where we set libtool for darwin right?

Re libtool/ar: http://www.manpages.info/macosx/libtool.1.html -> Libtool with -static is intended to replace ar(5) and ranlib. 
I know nothing more about this, and I don't feel strongly. It's just people that work on osx that tell me we should use libtool, so we do :)